src/main/java/nl/lexemmens/podman/AbstractUploadMojo.java

```diff
@@ -0,0 +1,152 @@
+package nl.lexemmens.podman;
+
+import org.apache.maven.RepositoryUtils;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.internal.impl.EnhancedLocalRepositoryManagerFactory;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.NoLocalRepositoryManagerException;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RepositoryPolicy;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class AbstractUploadMojo extends AbstractPodmanMojo {
```

Comment:
Maybe personal preference, but given the fact that this class also has a readLocal (and strictly speaking does not upload files), I'd opt to rename this to something like: `AbstractCatalogSupport`

Reply:
My initial reason for calling it `AbstractUploadMojo` was because both the CopyMojo and PushMojo are responsible for uploading files. But I can also see your reasoning to couple it more to the functional part. I'll change this to `AbstractCatalogSupport`.

Reply:
Updated